### DISCUSSION TITLE
Upgrading jdk from 16 to 17

### DISCUSSION THREAD

**Discussion by debjitpal5040** [microsoft/openjdk#163] [Q&A] [answered]
I have currently installed jdk 16 on my Mac M1 machine, I want to install jdk 17 on my device. So my question is if I install jdk 17 on my Mac, will it replace previously installed jdk 16 or they both will run parallel ? If they run parallel how could I remove jdk 16 and have only jdk 17 ?

**Comment by dsgrieve** [comment, marked as answer]
Your Mac will use the latest installed JVM. That said, you should check to see if you have JAVA_HOME set in your environment and update it accordingly. FWIW, your can use /usr/libexec/java_home to see what the current JVM is. Run it with -help for other options. You can toggle between versions with /usr/libexec/java_home --version <ver>, or set JAVA_HOME, e.g. `export JAVA_HOME=$(/usr/libexec/java_home --version 16)`

**Reply by debjitpal5040** [reply]
JAVA_HOME is not set in my environment as echo $JAVA_HOME outputs nothing. The JVM is also currently set to /Library/Java/JavaVirtualMachines/microsoft-16.jdk/Contents/Home. So installing JDK 17 will replace older JDK 16 installations and JVM ?

**Reply by dsgrieve** [reply]
Technically, 17 will install alongside of other JVMs. After install, 17 will be used. You can confirm with `/usr/libexec/java_home` (no args), and with `java -version`.

Personally, I don't bother with removing old versions. They come in handy some times and (on Mac) I can flip to an older version by setting JAVA_HOME. But if I was going to remove an older version, I'd install the new version first then just rm -rf the old version from /Library/Java/JavaVirtualMachines. YMMV

**Reply by debjitpal5040** [reply]
Thank you very much; cleared my doubt

**Reply by karianna** [reply]
You can also use the https://www.jenv.be/ tool to manage your versions
